CloudWatchSink/src/main/java/software/amazon/flink/example/sink/CloudWatchLogsSinkWriter.java

```diff
@@ -0,0 +1,183 @@
+                .collect(Collectors.toList());
+
+        PutLogEventsRequest request = PutLogEventsRequest.builder()
+                .logGroupName(logGroupName)
+                .logStreamName(logStreamName)
+                .logEvents(logsToSend.stream()
+                        .map(logEvent -> InputLogEvent.builder()
+                                .message(logEvent.getLog())
+                                .timestamp(logEvent.getTimestamp().toEpochMilli())
+                                .build())
+                        .collect(Collectors.toList()))
+                .build();
+
+        CompletableFuture<PutLogEventsResponse> future = cloudWatchLogsAsyncClient.putLogEvents(request);
+
+        log.debug("Sending batch of {} logs", logsToSend.size());
+
+        future.whenComplete((response, error) -> {
+            if (error == null) {
+                int successStartIndex = ofNullable(response.rejectedLogEventsInfo())
+                        .map(rejected -> Math.max(
+                                ofNullable(rejected.expiredLogEventEndIndex())
+                                        .orElse(0),
+                                ofNullable(rejected.tooOldLogEventEndIndex())
+                                        .orElse(0)))
+                        .orElse(0);
+
+                int successEndIndex = ofNullable(response.rejectedLogEventsInfo())
+                        .map(RejectedLogEventsInfo::tooNewLogEventStartIndex)
+                        .orElseGet(logsToSend::size);
+
+                // Re-enqueue the "too new" logs
+                failedLogs.accept(logsToSend.subList(successEndIndex, logsToSend.size()));
+
+                if (successStartIndex > 0) {
+                    log.warn("Failed to deliver {} too old/expired logs. {}",
+                            successEndIndex, logsToSend.subList(0, successStartIndex));
+                }
+
+                log.debug("Sent batch of {} logs", successEndIndex - successStartIndex);
+            } else {
+                log.warn("Failed to send batch of {} logs", logsToSend.size(), error);
+
+                if (isRetryable(error)) {
+                    failedLogs.accept(logsToSend);
+                }
+            }
+        });
+    }
+
+    @Override
+    protected long getSizeInBytes(LogEvent logEvent) {
+        // https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutLogEvents.html
+        // The maximum batch size is 1,048,576 bytes.
+        // This size is calculated as the sum of all event messages in UTF-8, plus 26 bytes for each log event.
+        // Assuming worst case 4 bytes per character
+
+        return logEvent.getLog().length() * 4L + 26;
+    }
+
+    @Override
+    public void close() {
+        asyncHttpClient.close();
+        cloudWatchLogsAsyncClient.close();
+    }
+
+    private boolean isRetryable(Throwable err) {
+        return FATAL_EXCEPTION_CLASSIFIER.isFatal(err, getFatalExceptionCons());
```

Comment:
Is this boolean backwards? I would think fatal exceptions == true should not be retried.